src/workerd/jsg/resource.h

```diff
@@ -458,6 +457,100 @@ JSG_DEFINE_GETTER_CALLBACK_STRUCTS(const)
+                                  method, isContext> { \
+      static constexpr bool enumerable = true; \
+      static void callback(const v8::FunctionCallbackInfo<v8::Value>& info) { \
+        liftKj(info, [&]() { \
+          auto isolate = info.GetIsolate(); \
+          auto context = isolate->GetCurrentContext(); \
+          auto obj = info.This(); \
+          auto& wrapper = TypeWrapper::from(isolate); \
+          /* V8 no longer supports AccessorSignature, so we must manually verify `this`'s type. */\
+          if (!isContext && !wrapper.template getTemplate(isolate, (T*)nullptr)->HasInstance(obj)) { \
+            throwTypeError(isolate, "Illegal invocation"); \
+          } \
+          auto& self = extractInternalPointer<T, isContext>(context, obj); \
+          return wrapper.wrap(context, obj, (self.*method)( \
+              wrapper.unwrap(context, (kj::Decay<Args>*)nullptr)...)); \
+        }); \
+      } \
+    }; \
+    \
+    /* Specialization for methods that take `Lock&` as their first parameter. */ \
+    template <typename TypeWrapper, const char* methodName, typename T, typename Ret, \
+              typename... Args, \
+              Ret (T::*method)(Lock&, Args...) __VA_ARGS__, \
+              bool isContext> \
+    struct PropertyGetterCallback<TypeWrapper, methodName, \
+                          Ret (T::*)(Lock&, Args...) __VA_ARGS__, \
+                          method, isContext> { \
+      static constexpr bool enumerable = true; \
+      static void callback(const v8::FunctionCallbackInfo<v8::Value>& info) { \
+        liftKj(info, [&]() { \
+          auto isolate = info.GetIsolate(); \
+          auto context = isolate->GetCurrentContext(); \
+          auto obj = info.This(); \
+          auto& wrapper = TypeWrapper::from(isolate); \
+          /* V8 no longer supports AccessorSignature, so we must manually verify `this`'s type. */\
+          if (!isContext && !wrapper.template getTemplate(isolate, (T*)nullptr)->HasInstance(obj)) { \
+            throwTypeError(isolate, "Illegal invocation"); \
+          } \
+          auto& self = extractInternalPointer<T, isContext>(context, obj); \
+          return wrapper.wrap(context, obj, (self.*method)(Lock::from(isolate), \
+              wrapper.unwrap(context, (kj::Decay<Args>*)nullptr)...)); \
+        }); \
+      } \
+    }; \
+    \
+    /* Specialization for methods that take `const v8::PropertyCallbackInfo<v8::Value>&` as \
+      * their first parameter. */ \
+    template <typename TypeWrapper, const char* methodName, typename T, typename Ret, \
+              typename... Args, \
+              Ret (T::*method)(const v8::PropertyCallbackInfo<v8::Value>&, Args...) __VA_ARGS__, \
+              bool isContext> \
+    struct PropertyGetterCallback<TypeWrapper, methodName, \
+                          Ret (T::*)(const v8::FunctionCallbackInfo<v8::Value>&, Args...) __VA_ARGS__, \
```

Comment:
Note for reviewers: with the change to url.h included in this PR, I don't believe we actually have any more cases where a property getter takes either a `v8::FunctionCallbackInfo` or a `v8::PropertyCallbackInfo`. So perhaps we can simplify things by removing these specializations?

Reply:
I'd be supportive of this!